### PR TITLE
Fix a few instances of signature mismatches

### PR DESCRIPTION
**Pull request by whymarrh**
This PR fixes a few instances of signature mismatches, where the args passed into function calls were either the wrong type or not needed. I've also updated two instances of JSDoc parameter types for optional params.

### CODE REVIEW

File: app/scripts/controllers/preferences.js [hunk 1]
```diff
@@ -671,7 +671,7 @@ class PreferencesController {
   /**
    * A getter for `tokens` and `accountTokens` related states.
    *
-   * @param {string} selectedAddress A new hex address for an account
+   * @param {string=} selectedAddress A new hex address for an account
```
Comment (Gudahtt) [inline app/scripts/controllers/preferences.js]: Interesting - you prefer the Closure syntax? The JSDoc syntax is more common in this repo:
```suggestion
   * @param {string} [selectedAddress] A new hex address for an account
```

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: app/scripts/controllers/preferences.js (hunk 1 above)
Comment (whymarrh) [reply app/scripts/controllers/preferences.js]: Nah, this is just a random decision made in the moment—I guess I know Closure syntax off-hand better than the JSDoc syntax.

File: app/scripts/controllers/transactions/lib/tx-state-history-helper.js [hunk 2]
```diff
@@ -35,7 +35,7 @@ function migrateFromSnapshotsToDiffs (longHistory) {
   with the first entry having the note and a timestamp when the change took place
   @param previousState {object} - the previous state of the object
   @param newState {object} - the update object
-  @param note {string} - a optional note for the state change
+  @param note {string=} - a optional note for the state change
```
Comment (Gudahtt) [inline app/scripts/controllers/transactions/lib/tx-state-history-helper.js]: Here's another case of non-conforming JSDoc syntax:
```suggestion
  @param {string} [note] - a optional note for the state change
```

**Review by Gudahtt — APPROVED** [review]

File: app/scripts/controllers/transactions/lib/tx-state-history-helper.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ function migrateFromSnapshotsToDiffs (longHistory) {
   with the first entry having the note and a timestamp when the change took place
   @param previousState {object} - the previous state of the object
   @param newState {object} - the update object
-  @param note {string} - a optional note for the state change
+  @param {string} [note] - a optional note for the state change
```
Comment (frankiebee) [inline app/scripts/controllers/transactions/lib/tx-state-history-helper.js]: what do the `[]` brackets do?

**Review by frankiebee — APPROVED** [review]

File: app/scripts/controllers/transactions/lib/tx-state-history-helper.js (hunk 3 above)
Comment (whymarrh) [reply app/scripts/controllers/transactions/lib/tx-state-history-helper.js]: They denote an optional parameter

**Pull request merged**